src/Dofs/DirichletBoundaryConditions.jl

```diff
@@ -0,0 +1,237 @@
+export DirichletBoundaryConditions, update!, apply!, apply_zero!, add!
+
+"""
+    DirichletBoundaryConditions
+
+A dirichlet boundary conditions is a a boundary where a dof is fixed to take a certain value.
+The struct `DirichletBoundaryConditions` represents a collection of such boundary conditions.
+
+It is created from a `DofHandler`
+
+```jldoctest dbc
+julia> dbc = DirichletBoundaryConditions(dh)
+```
+
+Dirichlet boundary conditions are added to certain components of a field for a specific nodes of the grid.
+A function is also given that should be of the form `(x,t) -> v` where `x` is the coordinate of the node, `t` is a
+time parameter and `v` should be of the same length as the number of components the bc is applied to:
+
+```jldoctest
+julia> addnodeset!(grid, "clamped", x -> norm(x[1]) ≈ 0.0);
+
+julia> nodes = grid.nodesets["clamped"]
+
+julia> push!(dbc, :temperature, nodes, (x,t) -> t * [x[2], 0.0, 0.0], [1, 2, 3])
+```
+
+Boundary conditions are now updates by specifying a time:
+
+```jldoctest
+julia> t = 1.0;
+
+julia> update!(dbc, t)
+```
+
+The boundary conditions can be applied to a vector:
+
+```jldoctest
+julia> u = zeros(ndofs(dh))
+
+julia> apply!(u, dbc)
+```
+
+"""
+immutable DirichletBoundaryCondition
+    f::Function
+    nodes::Set{Int}
+    field::Symbol
+    components::Vector{Int}
+    idxoffset::Int
+end
+
+
+immutable DirichletBoundaryConditions{DH <: DofHandler}
+    bcs::Vector{DirichletBoundaryCondition}
+    dofs::Vector{Int}
+    free_dofs::Vector{Int}
+    values::Vector{Float64}
+    dh::DH
+    closed::Ref{Bool}
+end
+
+function DirichletBoundaryConditions(dh::DofHandler)
+    @assert isclosed(dh)
+    DirichletBoundaryConditions(DirichletBoundaryCondition[], Int[], Int[], Float64[], dh, Ref(false))
+end
+
+function Base.show(io::IO, dbcs::DirichletBoundaryConditions)
+    println(io, "DirichletBoundaryConditions:")
+    if !isclosed(dbcs)
+        print(io, "  Not closed!")
+    else
+        println(io, "  BCs:")
+        for dbc in dbcs.bcs
+            println(io, "    ", "Field: ", dbc.field, " ", "Components: ", dbc.components)
+        end
+    end
+end
+
+isclosed(dbcs::DirichletBoundaryConditions) = dbcs.closed[]
+dirichlet_dofs(dbcs::DirichletBoundaryConditions) = dbcs.dofs
+free_dofs(dbcs::DirichletBoundaryConditions) = dbcs.free_dofs
+function close!(dbcs::DirichletBoundaryConditions)
```

Comment:
I thought `close!` like "the door is closed so you can't add anything more now. The fact that a bunch of things are happening under the scene doesn't really concern a user.